src/herder/test/HerderTests.cpp

```diff
@@ -665,6 +665,443 @@ TEST_CASE("txset", "[herder][txset]")
+                // gap between a1's seqLedger and lcl
+                uint32_t minGap;
+                if (v3ExtIsSet)
+                {
+                    // run a v19 op so a1's seqLedger is set
+                    a1.bumpSequence(0);
+                    closeLedger(*app);
+                    closeLedger(*app);
+                    minGap = 2;
+
+                    txSet = std::make_shared<TxSetFrame>(
+                        app->getLedgerManager()
+                            .getLastClosedLedgerHeader()
+                            .hash);
+                }
+                else
+                {
+                    // a1 seqLedger is 0 because it has not done a
+                    // v19 tx yet
+                    minGap = app->getLedgerManager().getLastClosedLedgerNum();
+                }
+
+                auto txInvalid = transactionWithV2Precondition(
+                    *app, a1, 1, 100, minSeqLedgerGapCond(minGap + 2));
+                txSet->add(txInvalid);
+                REQUIRE(!txSet->checkValid(*app, 0, 0));
+                REQUIRE(txSet->trimInvalid(*app, 0, 0).back() == txInvalid);
+                REQUIRE(txSet->sizeTx() == 0);
+
+                // we use minGap lcl + 1 because validation is done against the
+                // next ledger
+                auto tx1 = transactionWithV2Precondition(
+                    *app, a1, 1, 100, minSeqLedgerGapCond(minGap + 1));
+
+                // only the first tx can have minSeqLedgerGap set
+                auto tx2Invalid = transactionWithV2Precondition(
+                    *app, a1, 2, 100, minSeqLedgerGapCond(minGap + 1));
+
+                auto fb1 = feeBump(*app, a1, tx1, 200);
+                auto fb2Invalid = feeBump(*app, a1, tx2Invalid, 200);
+
+                if (minSeqNumTxIsFeeBump)
+                {
+                    txSet->add(fb1);
+                    txSet->add(fb2Invalid);
+                }
+                else
+                {
+                    txSet->add(tx1);
+                    txSet->add(tx2Invalid);
+                }
+
+                txSet->sortForHash();
+                REQUIRE(!txSet->checkValid(*app, 0, 0));
+
+                auto invalid = txSet->trimInvalid(*app, 0, 0);
+                REQUIRE(invalid.size() == 1);
+                REQUIRE(invalid.back() ==
+                        (minSeqNumTxIsFeeBump ? fb2Invalid : tx2Invalid));
+
+                REQUIRE(txSet->checkValid(*app, 0, 0));
+            };
+
+            SECTION("before v3 ext is set")
+            {
+                test(false, false);
+            }
+            SECTION("after v3 ext is set")
+            {
+                test(true, false);
+            }
+            SECTION("after v3 ext is set - fee bump")
+            {
+                test(true, true);
+            }
+        }
+        SECTION("minSeqAge")
+        {
+            auto minSeqAgeCond = [](uint32_t minSeqAge) {
```

Comment:
getting a warning on call sites because it's called with `Duration`

Reply:
Fixed